diff -ur libspnav-0.2.2/configure libspnav/configure
--- libspnav-0.2.2/spnav.h.orig	2011-11-30 22:54:06.447272302 +0100
+++ libspnav/spnav.h	2011-11-30 22:54:25.730605731 +0100
@@ -27,7 +27,7 @@
 #ifndef SPACENAV_H_
 #define SPACENAV_H_
 
-#include <spnav_config.h>
+#include "spnav_config.h"
 
 #ifdef USE_X11
 #include <X11/Xlib.h>
--- libspnav-0.2.2/configure.orig	2009-07-21 00:17:15.000000000 +0200
+++ libspnav-0.2.2/configure	2013-02-11 20:13:32.392810184 +0100
@@ -8,11 +8,11 @@
 X11=yes
 
 srcdir="`dirname "$0"`"
-libdir=lib
+libdir=$PREFIX/lib
 
-#if [ "`uname -m`" = 'x86_64' ]; then
-#	libdir=lib64
-#fi
+if [ "`uname -m`" = 'x86_64' ]; then
+	libdir=$PREFIX/lib64
+fi
 
 for arg; do
 	case "$arg" in
@@ -20,7 +20,10 @@
 		value=`echo $arg | sed 's/--prefix=//'`
 		PREFIX=${value:-$prefix}
 		;;
-	
+	--libdir=*)
+		libdir=`echo $arg | sed 's/--libdir=//'`
+		;;
+
 	--enable-opt)
 		OPT=yes;;
 	--disable-opt)
--- libspnav-0.2.3/Makefile.in.orig	2014-08-18 03:38:31.000000000 +0200
+++ libspnav-0.2.3/Makefile.in	2017-12-17 18:07:21.386501353 +0100
@@ -34,7 +34,7 @@
 	$(AR) rcs $@ $(obj)
 
 $(lib_so): $(obj)
-	$(CC) $(shared) -o $@ $(obj) $(LDFLAGS)
+	$(CC) $(LDFLAGS) $(shared) -o $@ $(obj)
 
 %.o: $(srcdir)/%.c
 	$(CC) $(CFLAGS) -c $< -o $@
@@ -49,22 +49,22 @@
 
 .PHONY: install
 install: $(lib_a) $(lib_so)
-	mkdir -p $(DESTDIR)$(PREFIX)/$(libdir) $(DESTDIR)$(PREFIX)/include
-	cp $(lib_a) $(DESTDIR)$(PREFIX)/$(libdir)/$(lib_a)
-	cp $(lib_so) $(DESTDIR)$(PREFIX)/$(libdir)/$(lib_so)
+	mkdir -p $(DESTDIR)$(libdir) $(DESTDIR)$(PREFIX)/include
+	cp $(lib_a) $(DESTDIR)$(libdir)/$(lib_a)
+	cp $(lib_so) $(DESTDIR)$(libdir)/$(lib_so)
 	[ -n "$(soname)" ] && \
-		rm -f $(DESTDIR)$(PREFIX)/$(libdir)/$(soname) $(DESTDIR)$(PREFIX)/$(libdir)/$(devlink) && \
-		ln -s $(DESTDIR)$(PREFIX)/$(libdir)/$(lib_so) $(DESTDIR)$(PREFIX)/$(libdir)/$(soname) && \
-		ln -s $(DESTDIR)$(PREFIX)/$(libdir)/$(soname) $(DESTDIR)$(PREFIX)/$(libdir)/$(devlink) || \
+		rm -f $(DESTDIR)$(libdir)/$(soname) $(DESTDIR)$(libdir)/$(devlink) && \
+		ln -s $(lib_so) $(DESTDIR)$(libdir)/$(soname) && \
+		ln -s $(soname) $(DESTDIR)$(libdir)/$(devlink) || \
 		true
 	for h in $(hdr); do cp -p $(srcdir)/$$h $(DESTDIR)$(PREFIX)/include/; done
 
 .PHONY: uninstall
 uninstall:
-	rm -f $(DESTDIR)$(PREFIX)/$(libdir)/$(lib_a)
-	rm -f $(DESTDIR)$(PREFIX)/$(libdir)/$(lib_so)
+	rm -f $(DESTDIR)$(libdir)/$(lib_a)
+	rm -f $(DESTDIR)$(libdir)/$(lib_so)
 	[ -n "$(soname)" ] && \
-		rm -f $(DESTDIR)$(PREFIX)/$(libdir)/$(soname) $(DESTDIR)$(PREFIX)/$(libdir)/$(devlink) || \
+		rm -f $(DESTDIR)$(libdir)/$(soname) $(DESTDIR)$(libdir)/$(devlink) || \
 		true
 	for i in $(hdr); do rm -f $(DESTDIR)$(PREFIX)/include/$$i; done
 
